augur/utils.py

```diff
@@ -175,6 +175,68 @@ def load_features(reference, feature_names=None):
     else:
         return _read_genbank(reference, feature_names)
 
+def _read_nuc_annotation_from_gff(record, reference):
+    """
+    Looks for the 'record' & 'source' GFF type as well as the ##sequence-region
+    pragma. Note that 'source' isn't really a GFF feature type, but is used
+    widely in the Nextstrain ecosystem. If there are multiple we check that the
+    coordinates agree.
+    
+    Parameters
+    ----------
+    record : <class 'Bio.SeqRecord.SeqRecord'> reference: string
+
+    Returns
+    -------
+    <class 'Bio.SeqFeature.SeqFeature'>
+
+    Raises
+    ------
+    AugurError
+        If no information on the genome / seqid length is available or if the
+        information is contradictory
+    """
+    nuc = {}
+    # Attempt to parse the sequence-region pragma to learn the genome
+    # length (in the absence of record/source we'll use this for 'nuc')
+    sequence_regions = record.annotations.get('sequence-region', [])
+    if len(sequence_regions)>1:
+        raise AugurError(f"Reference {reference!r} contains multiple ##sequence-region pragma lines. Augur can only handle GFF files with a single one.")
+    elif sequence_regions:
+        from Bio.SeqFeature import SeqFeature, FeatureLocation
+        (name, start, stop) = sequence_regions[0]
+        nuc['pragma'] = SeqFeature(
+            FeatureLocation(start, stop),
+            strand=1,
+            type='##sequence-region pragma',
+            id=name,
+        )
+    for feat in record.features:
+        if feat.type == "region":
+            nuc['region'] = feat
+        elif feat.type == "source":
+            nuc['source'] = feat
+
+    if not nuc:
+        raise AugurError(f"Reference {reference!r} didn't define any information we can use to create the 'nuc' annotation. You can use a line with a 'record' or 'source' GFF type or a ##sequence-region pragma.")
+
+    # ensure they all agree on coordinates, if there are multiple
+    if len(nuc.values())>1:
```

Comment:
Is there a reason we want to look for all possible "nuc" coordinates and then check that they match when we only return the coordinates we prefer in the end? What if we just returned the first nuc coordinates we find in our preferred order?

Reply:
My thinking is that if there's disagreement you've got bigger problems with your GFF and should go fix it. We don't know which (if any) of the disagreeing coordinates are the "correct" ones.